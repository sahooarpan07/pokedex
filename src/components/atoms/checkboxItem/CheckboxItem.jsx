import {View, Text} from 'react-native';
import React from 'react';
import {styles} from './CheckboxItem.styles';
import CheckBox from '@react-native-community/checkbox';
import {getCapitalisedWord} from '../../../util/utils';

const CheckboxItem = ({value, onValueChange, label}) => {
  return (
    <View style={styles.container}>
      <CheckBox
        disabled={false}
        value={value}
        onValueChange={newValue => onValueChange(newValue)}
        testID="checkbox"
      />
      <Text style={styles.label}>{getCapitalisedWord(label)}</Text>
    </View>
  );
};

export default CheckboxItem;
