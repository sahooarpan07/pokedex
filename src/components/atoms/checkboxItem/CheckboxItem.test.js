import 'react-native';
import React from 'react';
import CheckboxItem from './CheckboxItem';
import {render, fireEvent, screen} from '@testing-library/react-native';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';

test('should render Checkbox with label', () => {
  const tree = renderer
    .create(
      <CheckboxItem value={true} label={'Grass'} onValueChange={jest.fn()} />,
    )
    .toJSON();

  expect(tree).toMatchSnapshot();
});
