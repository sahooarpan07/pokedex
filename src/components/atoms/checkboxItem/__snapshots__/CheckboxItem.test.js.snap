// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Checkbox with label 1`] = `
<View
  style={
    {
      "flexDirection": "row",
    }
  }
>
  <View
    accessibilityRole="checkbox"
    accessibilityState={
      {
        "checked": true,
        "disabled": false,
      }
    }
    accessibilityValue={
      {
        "text": "on",
      }
    }
    accessible={true}
    pointerEvents="auto"
    testID="checkbox"
  >
    <RNCCheckbox
      forwardedRef={null}
      onValueChange={[Function]}
      style={
        [
          {
            "backgroundColor": "transparent",
            "height": 32,
            "width": 32,
          },
          undefined,
        ]
      }
      value={true}
    />
  </View>
  <Text
    style={
      {
        "fontWeight": "bold",
        "marginTop": 4,
      }
    }
  >
    Grass
  </Text>
</View>
`;
